File: 006/Problem6.go
package main

import (
	"fmt"
	"time"
)

func sumOfSquares(num int) int {
	sum := 0
	for num > 0 {
		sum += num * num
		num--
	}
	return sum
}

func squareOfSum(num int) int {
	sum := 0
	for num > 0 {
		sum += num
		num--
	}
	return sum * sum
}

func problem() {
	naturalNum := 100
	fmt.Println(squareOfSum(naturalNum) - sumOfSquares(naturalNum))
}

func main() {
	start := time.Now()
	for i := 0; i < 1000; i++ {
		problem()
	}
	elapsed := time.Since(start)
	fmt.Printf("Function took %s", elapsed)
}

File: 006/Problem6.py
import time


def SumOfSquares(num: int) -> int:
    su = 0
    for i in range(0, num + 1):
        su += i * i
    return su


def SquareOfSum(num: int) -> int:
    su = 0
    for i in range(0, num + 1):
        su += i
    return su * su


def problem():
    print(SquareOfSum(100) - SumOfSquares(100))


start = time.time()
for i in range(1000):
    problem()
end = time.time()
print("Function took " + str(end - start))
